semanticcpg/src/test/scala/io/shiftleft/semanticcpg/passes/CfgDominatorFrontierTests.scala

```diff
@@ -23,7 +23,7 @@ class CfgDominatorFrontierTests extends AnyWordSpec with Matchers {
       node.in("CFG").asScala
   }
 
-  private class TestDomTreeAdapter(immediateDominators: Map[Node, Node]) extends DomTreeAdapter[Node] {
+  private class TestDomTreeAdapter(immediateDominators: scala.collection.Map[Node, Node]) extends DomTreeAdapter[Node] {
```

Comment:
was that change intended?

Reply:
Yes. `Map` by default refers to `scala.collection.immutable.Map`, whereas `scala.collection.Map` is a shared super of both mutable and immutable maps. I consider that design decision of the scala team rather questionable.

In order to ensure determinism, I switched the actual map out to `scala.collection.mutable.LinkedHashMap`, so this change was necessary if we want to avoid unnecessary copies.

Reply:
👍🏻